src/httpd_streaming.c

```diff
@@ -216,55 +297,56 @@ streaming_meta_cb(evutil_socket_t fd, short event, void *arg)
   if (!decode_ctx)
     goto error;
 
-  streaming_encode_ctx = transcode_encode_setup(XCODE_MP3, &streaming_quality_out, decode_ctx, NULL, 0, 0);
+  ctx->encode_ctx = transcode_encode_setup(ctx->xcode, &ctx->quality_out, decode_ctx, NULL, 0, 0);
   transcode_decode_cleanup(&decode_ctx);
-  if (!streaming_encode_ctx)
+  if (!ctx->encode_ctx)
     {
-      DPRINTF(E_LOG, L_STREAMING, "Will not be able to stream MP3, libav does not support MP3 encoding: %d/%d/%d @ %d\n", streaming_quality_out.sample_rate, streaming_quality_out.bits_per_sample, streaming_quality_out.channels, streaming_quality_out.bit_rate);
-      streaming_not_supported = 1;
+      DPRINTF(E_LOG, L_STREAMING, "Will not be able to stream %s, libav does not support encoding: %d/%d/%d @ %d\n", ctx->name, ctx->quality_out.sample_rate, ctx->quality_out.bits_per_sample, ctx->quality_out.channels, ctx->quality_out.bit_rate);
+      ctx->not_supported = 1;
+      streaming_end(ctx);
```

Comment:
The addition of `streaming_end(..)` is a bug fix - if we don't have this, the `sessions` list is never cleared down so the player will continue to send data and we reenter here.